### ISSUE TITLE
Multigroup Estimation

### ISSUE THREAD

**Issue by oguzhanogreden** [oguzhanogreden/mirt#14]
I decided to remove `mean=` and `sd=` from `dcurver` function interface,  to match the publications of Woods.

I'll follow this [here](https://github.com/oguzhanogreden/dcurver/issues/7). Means will be fine but SDs I want to check first.

**Comment by philchalmers** [comment]
I had this on my TODO list as well and have been chewing over what exactly to do in the interface. I  think including these as parameters in the `GroupPars` object makes any sense if the mean and sd's influence can be included in the `ddc` density components and gradients/Hessian. 

The `standardizeQuadrature()` function would also need to accept the previous iterations mean and sd terms for rescaling too....but that's the easy part.

**Comment by oguzhanogreden** [comment]
I thought this should be possible since the Davidian curve is `P_k^2(theta) * N(theta)` where `N(theta)` is standard normal density. Then, I thought, I could move the `N(theta)` around and have the multigroup case.

But the parameterization which ensures this integrates to one is tied to the standard normal case, see Equations 7 through 9 in Woods & Lin (2009). Also, you can use [this branch](https://github.com/oguzhanogreden/dcurver/tree/allow-mean-sd) to confirm:

```
integrate(ddc, -20, 20, mean = 0, sd = 1, phi=runif(5, 1, 5)) # == 1
integrate(ddc, -20, 20, mean = 0.1, sd = 1, phi=runif(5, 1, 5)) # != 1
```

I guess that the required modification is simply expressing the `M` matrix in Equation 9 in a more general way. I would need to study a while to formally confirm and attempt at doing this. I'd learn a lot so I'd like to try this, perhaps in summer.

This is a somewhat interesting bit of the parameterization in any case, let me also note that this M matrix has elements up to `E(Z^18)` where `Z ~ N(0,1)` for `len(phi) = 10`. Currently, dcurver uses a crude approximation for this.

**Comment by philchalmers** [comment]
I see. That's a bit of a problem for now then for now. The real benefit of multiple group estimation is to equate the groups via anchor items, which allows for the latent variable scales to be shifted via freeing the focal group(s) mean and variance in $\theta$. If we can't find a reasonable way to do this with DCs that's unfortunate, but at least the groundwork has been laid for the Ramsey curve approach (at some point...this is a less pressing feature).

**Comment by philchalmers** [comment]
A placeholder for this method is currently working (sans equality constraints for now), though I'm not convinced it works for multiple-group models with freely estimated mean/vars in the focal groups. It seems to work fine when the [0,1] constraint is imposed in both groups though, so at least the logic works. It's likely this should go onto a delayed TODO list until suitable mean/variance parameters can be estimated for DCs.

**Comment by oguzhanogreden** [comment]
I agree with the delayed TODO idea. This is not a priority for me at the moment and it will likely be a major part of one of my next projects.

**Comment by philchalmers** [comment]
Sounds good. I'll throw a stop() for now then whenever the latent mean/var parameters are set to TRUE, and we can leave this issue open until the estimation of these parameters is resolved at a future date.
